src/PIL/PpmImagePlugin.py

```diff
@@ -49,71 +53,264 @@ class PpmImageFile(ImageFile.ImageFile):
+            b"P2": "image/x-portable-graymap",
+            b"P3": "image/x-portable-pixmap",
             b"P4": "image/x-portable-bitmap",
             b"P5": "image/x-portable-graymap",
             b"P6": "image/x-portable-pixmap",
         }.get(magic_number)
 
-        if mode == "1":
-            self.mode = "1"
-            rawmode = "1;I"
-        else:
-            self.mode = rawmode = mode
-
         for ix in range(3):
-            while True:
-                while True:
-                    s = self.fp.read(1)
-                    if s not in b_whitespace:
-                        break
-                    if s == b"":
-                        raise ValueError("File does not extend beyond magic number")
-                if s != b"#":
-                    break
-                s = self.fp.readline()
-            s = int(self._token(s))
-            if ix == 0:
-                xsize = s
-            elif ix == 1:
-                ysize = s
+            token = self._read_token()
+            try:  # check token sanity
+                token = int(token)
+            except ValueError:
+                raise ValueError(
+                    f"Non-decimal-ASCII found in header: {token}"
+                ) from None
+            if ix == 0:  # token is the x size
+                xsize = token
+            elif ix == 1:  # token is the y size
+                ysize = token
                 if mode == "1":
+                    self.mode = "1"
+                    rawmode = "1;I"
                     break
-            elif ix == 2:
-                # maxgrey
-                if s > 255:
+                else:
+                    self.mode = rawmode = mode
+            elif ix == 2:  # token is maxval
+                maxval = token
+                if maxval > 255:
                     if not mode == "L":
-                        raise ValueError(f"Too many colors for band: {s}")
-                    if s < 2 ** 16:
+                        raise ValueError(f"Too many colors for band: {maxval}")
+                    if maxval < 2 ** 16:
                         self.mode = "I"
                         rawmode = "I;16B"
                     else:
                         self.mode = "I"
                         rawmode = "I;32B"
 
         self._size = xsize, ysize
-        self.tile = [("raw", (0, 0, xsize, ysize), self.fp.tell(), (rawmode, 0, 1))]
+        self.tile = [
+            (
+                decoder,  # decoder
+                (0, 0, xsize, ysize),  # region: whole image
+                self.fp.tell(),  # offset to image data
+                (rawmode, 0, 1),  # parameters for decoder
+            )
+        ]
+
+
+#
+# --------------------------------------------------------------------
+
+
+class PpmPlainDecoder(ImageFile.PyDecoder):
+    _pulls_fd = True
+
+    def _read_block(self, block_size=10 ** 6):
+        return bytearray(self.fd.read(block_size))
+        # return self.fd.read(block_size)
```

Comment:
```suggestion
```
Is this unintentionally left in? Or were you trying to communicate something to future developers?

Reply:
I was unintentional, sorry.